packages/ui/src/machines/authenticator/__tests__/defaultService.test.ts

```diff
@@ -0,0 +1,105 @@
+import { PasswordSettings } from '../../../types';
+import { defaultServices } from '../defaultServices';
+
+const { validateFormPassword } = defaultServices;
+
+const untouched = { password: false };
+const touched = { password: true };
+
+const strictPasswordPolicy: PasswordSettings = {
+  passwordPolicyCharacters: [
+    'REQUIRES_LOWERCASE',
+    'REQUIRES_NUMBERS',
+    'REQUIRES_UPPERCASE',
+    'REQUIRES_SYMBOLS',
+  ],
+  passwordPolicyMinLength: 8,
+};
+
+const lenientPasswordPolicy: PasswordSettings = {
+  passwordPolicyCharacters: [],
+  passwordPolicyMinLength: 4,
+};
+
+describe('validateFormPassword', () => {
+  it('happy case with valid password and strict password policy', async () => {
+    const password = 'UnitTest_Password1';
+    const result = await validateFormPassword({ password }, touched, {
+      passwordPolicyCharacters: ['REQUIRES_SYMBOLS'],
+      passwordPolicyMinLength: 1,
+    });
+    expect(result).toBe(null);
+  });
+
+  it('unhappy cases with invalid passwords and strict password policy', async () => {
+    // has special character not recognized by Cognito
+    const password1 = 'UnitTestㄱPassword1';
+    const result1 = await validateFormPassword(
+      { password: password1 },
+      touched,
+      strictPasswordPolicy
+    );
+    expect(result1).toStrictEqual({
+      password: ['Password must have special characters'],
+    });
+
+    // does not have special character
+    const password2 = 'UnitTestPassword1';
+    const result2 = await validateFormPassword(
+      { password: password2 },
+      touched,
+      strictPasswordPolicy
+    );
+    expect(result2).toStrictEqual({
+      password: ['Password must have special characters'],
+    });
+
+    // is too short, and does not meet character requirements
+    const password3 = 'short';
+    const result3 = await validateFormPassword(
+      { password: password3 },
+      touched,
+      strictPasswordPolicy
+    );
+    expect(result3).toStrictEqual({
+      password: [
+        'Password must have at least 8 characters',
+        'Password must have numbers',
+        'Password must have upper case letters',
+        'Password must have special characters',
+      ],
+    });
+  });
+
+  it('happy case with valid password and lenient password policy', async () => {
```

Comment:
Nit: use names that read starting with `it`, example
```
it('validates as expected with valid password and lenient password policy')
```

Reply:
Oh gosh, I finally realize what `it` stands for.

Reply:
[1336cb0](https://github.com/aws-amplify/amplify-ui/pull/2071/commits/1336cb0747003cd4167359498ae19c5937da5458)

Reply:
I made few more wording commits after [1336cb0](https://github.com/aws-amplify/amplify-ui/pull/2071/commits/1336cb0747003cd4167359498ae19c5937da5458)... I keep missing one by one!

I think it looks good now, will merge later today if no objections.